File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/StandupScreenData.java
/**
 * This file is part of mycollab-web.
 *
 * mycollab-web is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-web is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-web.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.project.domain.StandupReportWithBLOBs;
import com.esofthead.mycollab.module.project.domain.criteria.StandupReportSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

/**
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 *
 */
public class StandupScreenData {
	public static class Search extends ScreenData<StandupReportSearchCriteria> {

		public Search(StandupReportSearchCriteria params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<StandupReportWithBLOBs> {

		public Add(StandupReportWithBLOBs params) {
			super(params);
		}
	}

	public static class Edit extends ScreenData<StandupReportWithBLOBs> {

		public Edit(StandupReportWithBLOBs params) {
			super(params);
		}
	}

	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/scala/com.esofthead.mycollab.module.project.view.parameters/StandupScreenData.scala
package com.esofthead.mycollab.module.project.view.parameters

import java.util.Date

import com.esofthead.mycollab.module.project.domain.StandupReportWithBLOBs
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
object StandupScreenData {

  class Search(param: Date) extends ScreenData[Date](param) {}

  class Add(param: StandupReportWithBLOBs) extends ScreenData[StandupReportWithBLOBs](param) {}

  class Edit(param: StandupReportWithBLOBs) extends ScreenData[StandupReportWithBLOBs](param) {}

}
